server/server.go

```diff
@@ -352,33 +350,16 @@ func (s *Server) listenUpdates() (err error) {
 				}
 			}
 
-			conn := conn // Copy conn to a local variable
-			go func() {
```

Comment:
What's the reason to remove sending it in parallel?

Reply:
No real reason. I just figured doing it in parallel was overcomplicating it since it's processed sequentially anyway and we have to wait at the end for it. But I guess we do get a speedup by being able to push all the work to a channel without waiting. Will make it parallel again.

Reply:
Actually, one good thing about doing it sequentially is order is preserved. You can't guarantee the order things will be written to the channel if you write in parallel. For that reason I think doing it sequentially should be safer.

Reply:
Indeed I just tested what happens if I make it parallel again and use a waitgroup to wait for all goroutines and yep, the old behaviour returns. Doing it in parallel is dangerous.

Reply:
But even doing it that way deadlocks if it can't write the response (e.g. close email client). Back to the drawing board...

Reply:
> order is preserved
But order here doesn't matter as it's order which connection gets the data. We don't care which connection gets faster response...

> doing it sequentially should be safer
Each connection has it's own channel, why it should not be safe doing in parallel?

Reply:
Suppose there is only one channel. I send two idle updates quickly one after the other. We start two goroutines, each charged with sending one update. Who knows which one completes first... maybe the second one actually finishes first.

That's why I think writing to the update channel should always be done serially.